### PR TITLE
Ruff

### PR DESCRIPTION
**Pull request by rknop**
Add the ruff linter.   Put it in the github workflows before the tests already run.

You can run the ruff linter in your own checkout by just running `ruff check` in the top directory.  If you don't have ruff installed on your system, just go into the shell of the test environment and run it there.

### CODE REVIEW

File: util/util.py [hunk 1]
```diff
@@ -425,7 +426,7 @@ def get_inheritors(klass):
 
 
 def as_UUID( val, canbenone=True ):
-    """Convert a string or None to a uuid.UUID
+    r"""Convert a string or None to a uuid.UUID
```
Comment (whohensee) [inline util/util.py:429]: do we need this raw string tag?

File: util/util.py [hunk 2]
```diff
@@ -456,7 +457,7 @@ def as_UUID( val, canbenone=True ):
 
 
 def as_datetime( string ):
-    """Convert a string to datetime.date with some error checking, allowing a null op.
+    r"""Convert a string to datetime.date with some error checking, allowing a null op.
```
Comment (whohensee) [inline util/util.py:460]: same question on if the r-string is needed?

**Review by whohensee — COMMENTED** [review]
Looks good. I don't remember what my earlier comments were, but there was at least one error so keep an eye out for that.

I like the style that ruff is enforcing, so this should be good to have!

File: conductor/updater.py [hunk 3]
```diff
@@ -4,23 +4,21 @@
...
 import logging
 import json
-import multiprocessing
 
 # Have to manually import any instrument modules
 #  we want to be able to find.  (Otherwise, they
 #  won't be found when models.instrument  is
 #  initialized.)
-import models.decam
+import models.decam   # noqa: F401
```
[the remark this answers is not in the document]
Comment (whohensee) [reply conductor/updater.py:14]: I have since read the sections on adding ruff so of course this makes sense now

File: util/util.py (hunk 2 above)
Comment (rknop) [reply util/util.py:460]: This one *is* needed because of the backslashes further down in the docstring.

**Review by whohensee — APPROVED** [review]

**Pull request merged**